packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -539,14 +548,24 @@ class BallisticScrollActivity extends ScrollActivity {
 
   late AnimationController _controller;
 
+  late double _initVelocity;
+
+  late double _initPosition;
+
   @override
   void resetActivity() {
     delegate.goBallistic(velocity);
   }
 
   @override
   void applyNewDimensions() {
-    delegate.goBallistic(velocity);
+    final Simulation? newSimulation = delegate.updateBallisticAnimation(
+      _initVelocity,
+      _initPosition,
+    );
+    if (newSimulation != null) {
+      _controller.updateSimulation(newSimulation);
```

Comment:
How is it guaranteed that this doesn't give as a discontinued animation with a big jump when we switch simulations? (see my other comment about this above)